pyomo/contrib/parmest/parmest.py

```diff
@@ -613,19 +1036,169 @@ def _Q_opt(
+        """
+        # Number of data points considered
+        n = cov_n
+
+        # Extract number of fitted parameters
+        l = len(self.estimated_theta)
+
+        # Assumption: Objective value is sum of squared errors
+        sse = self.objective_value
+
+        '''Calculate covariance assuming experimental observation errors are
+        independent and follow a Gaussian distribution with constant variance.
+
+        The formula used in parmest was verified against equations (7-5-15) and
+        (7-5-16) in "Nonlinear Parameter Estimation", Y. Bard, 1974.
+
+        This formula is also applicable if the objective is scaled by a constant;
+        the constant cancels out. (was scaled by 1/n because it computes an
+        expected value.)
+        '''
+        # check if the supplied method is supported
+        try:
+            cov_method = CovMethodLib(method)
+        except:
+            raise ValueError(f"Invalid method: '{method}'. Choose from {[e.value for e in CovMethodLib]}.")
+
+        # get a version of the model to check if it has a `measurement_error` attribute
+        model = self.exp_list[0].get_labeled_model()
+
+        # check if the user specified `SSE` or `SSE_weighted` as the objective function
+        if self.obj_function == ObjectiveLib.SSE:
+            # check if user defined the `measurement_error` attribute
+            if hasattr(model, "measurement_error"):
+                # get the measurement errors
+                meas_error = [
+                    model.measurement_error[y_hat]
+                    for y_hat, y in model.experiment_outputs.items()
+                ]
+
+                # check if the user supplied the values of the measurement errors
+                if all(item is None for item in meas_error):
+                    measurement_var = sse / (
+                        n - l
+                    )  # estimate of the measurement variance
+                    if cov_method == CovMethodLib.reduced_hessian:
+                        cov = (
+                            2 * measurement_var * self.inv_red_hes
+                        )  # covariance matrix
+                        cov = pd.DataFrame(
+                            cov,
+                            index=self.estimated_theta.keys(),
+                            columns=self.estimated_theta.keys(),
+                        )
+                    elif cov_method == CovMethodLib.finite_difference or cov_method == CovMethodLib.kaug:
+                        cov = compute_cov(
+                            self.exp_list,
+                            method,
+                            thetavals=self.estimated_theta,
+                            solver=solver,
+                            step=step,
+                            tee=self.tee,
+                            estimated_var=measurement_var,
+                        )
+                    else:
+                        raise NotImplementedError(
```

Comment:
Perhaps define the NotImplementedError in one spot above? Or at least it's text.